### DISCUSSION TITLE
Unable to setup Milvus on AKS

### DISCUSSION THREAD

**Discussion by Az-dev-9** [milvus-io/milvus#28171] [Q&A and General discussion] [answered]
Hello Team,

We are trying to move away from Milvus on Standalone server and we are trying to deploy on Azure Kubernetes Services.

We are following the documentation from here https://milvus.io/docs/v2.1.x/azure.md.

However it looks like not all pods are not running correctly:

![image](https://github.com/milvus-io/milvus/assets/57792398/03a006d3-dd60-4771-80cd-6bf2a380a34f)

Even if we try to use Azure Storage account from the guide it looks like things are getting worse.

![image](https://github.com/milvus-io/milvus/assets/57792398/f448b521-551a-417d-8fd8-d99b562f87ce)

So no matter what we try we are unable to connect to Milvus through the Load Balancer it always times out.

![image](https://github.com/milvus-io/milvus/assets/57792398/98a7cbfa-369c-4271-bb49-69a408acdb41)

The issue is not related with the NSG as we checked this multiple times.

Is there something that we are missing when configuring? We are following the steps strictly and nothing works.

Any help with this will be appreciated. 

Best Regards,

Anton

**Comment by yhmo** [comment]
Logs of the datacoord/querynode would be helpful to know what happened.

**Comment by Az-dev-9** [comment]
Hi @yhmo , 

Thanks for taking a look at this.

I have attached the requested logs
[datacoordlogs.txt](https://github.com/milvus-io/milvus/files/13264758/datacoordlogs.txt)
[querynodelogs.txt](https://github.com/milvus-io/milvus/files/13264759/querynodelogs.txt)

I just want to clarify again that this is initial setup issue.

Please let me know if other details are needed

**Reply by yhmo** [reply]
The log of datacoord says "find no available rootcoord, check rootcoord state". Need the log of rootcoord.

**Reply by yhmo** [reply]
Or refer this [doc](https://github.com/milvus-io/milvus/tree/master/deployments/export-log) to export the entire Milvus logs.

**Reply by Az-dev-9** [reply]
Hi @yhmo I am sorry about the delay.
[MilvusPodsLogs.zip](https://github.com/milvus-io/milvus/files/13309511/MilvusPodsLogs.zip)
 Here are the logs from all (running) pods

**Reply by yhmo** [reply]
The rootcoord could not connect pulsar and exit with this error log:
```
time="2023-11-09T14:11:55Z" level=info msg="Connecting to broker" remote_addr="pulsar://my-release-pulsar-proxy:6650"
time="2023-11-09T14:11:55Z" level=warning msg="Failed to connect to broker." error="dial tcp 10.0.206.31:6650: connect: connection refused" remote_addr="pulsar://my-release-pulsar-proxy:6650"
```

But I didn't see the log of broker and proxy, the zip file only contains log of pulsar-bookie and pulsar-zookeeper. I guess the pulsar-broker and pulsar-proxy is unhealthy.

**Reply by Az-dev-9** [reply]
For some sort of reason it started working today... (I shut down the AKS every night while we are testing)

I will try to make another deployment to see if I will get the issue again

![image](https://github.com/milvus-io/milvus/assets/57792398/b63d99bc-c410-4d43-860f-e5fce3ac5987)

Is it normal for the pods to have a different ID

**Reply by Az-dev-9** [reply, marked as answer]
Hi @yhmo , @zwd1208 , 

I found why my Milvus on Kubernetes Cluster was able to start and why all new deployments are failing.

Actually the answer hidden in the screenshots and one of @yhmo findings!

Why are all "Pulsar" pods unable to start?

![image](https://github.com/milvus-io/milvus/assets/57792398/4d19bbe0-34ad-4be4-a953-791ecba396f3)


Because there is a missing Inbound NSG rule with the initial deployment. Once I allowed port 6650 milvus started working (I needed to restart my AKS as well)

![image](https://github.com/milvus-io/milvus/assets/57792398/bcb93633-d5aa-4f8d-a057-ebab5988d0e5)

I found in your documentation that this port is being used by Pulsar https://milvus.io/docs/v2.0.x/configure_pulsar.md 

I am not sure why your Azure Deployment is not allowing it, as it is doing with port 19530 and 9091 but I think you should consider adding this in you documentation: port 6650 needs to added in your Network Security group when using Milvus on AKS.

I hope that this will be helpful to others I found that someone else was facing the exact same issue some months ago https://learn.microsoft.com/en-us/answers/questions/1305931/operation-timed-out-for-an-external-ip-in-azure-cl?source=docs 

Best Regards 

Anton

**Comment by zwd1208** [comment]
@Az-dev-9 can you describe the none running pods by kubectl. use `kubectl describe pod <podname> -n <namespace>`
